File: backend/repair_service/schemas/diagnosis.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, List

class DiagnosisBase(BaseModel):
    diagnosis_id: int = Field(..., description='Mã chẩn đoán')
    form_id: int = Field(..., description='Mã phiếu sửa chữa')
    order_id: int = Field(..., description='Mã đơn hàng')
    problem: str = Field(..., description='Vấn đề phát hiện')
    created_at: datetime = Field(default= datetime.now(), description='Ngày chẩn đoán')
    estimated_cost: int = Field(..., description='Chi phí ước tính')

class DiagnosisCreate(BaseModel):
    form_id: int = Field(..., description='Mã phiếu sửa chữa')
    order_id: int = Field(..., description='Mã đơn hàng')
    class Config:
        from_attributes = True
        json_schema_extra = {
            "example": {
                "form_id": 1,
                "order_id": 1
            }
        }

class DiagnosisUpdate(BaseModel):
    problem: Optional[str]
    estimated_cost: Optional[int]
    class Config:
        from_attributes = True
        json_schema_extra = {
            "example": {
                "problem": "Xe không khởi động được",
                "estimated_cost": 500000
            }
        }

class DiagnosisResponse(BaseModel):
    diagnosis_id: int
    form_id: int
    order_id: int
    problem: Optional[str]
    created_at: datetime
    estimated_cost: Optional[int]

    class Config:
        from_attributes = True
        json_schema_extra = {
            "example": {
                "diagnosis_id": 1,
                "form_id": 1,
                "order_id": 1,
                "created_at": datetime.now(),
            }
        }
# class DiagnosisInDB(DiagnosisBase):
#     diagnosis_id: int
#     created_at: datetime
    
#     class Config:
#         orm_mode = True

# class Diagnosis(DiagnosisInDB):
#     pass

# class DiagnosisWithOrders(Diagnosis):
#     orders: List = []

File: frontend/src/models/Diagnosis.js
import BaseModel from './BaseModel';

export default class Diagnosis extends BaseModel {
  constructor(data = {}) {
    super(data);
  }

  /**
   * Get diagnosis ID
   */
  get id() {
    return this._data.diagnosis_id;
  }

  /**
   * Get problem description
   */
  get problem() {
    return this._data.problem;
  }

  /**
   * Set problem description
   */
  set problem(value) {
    this._data.problem = value;
  }

  /**
   * Get estimated cost
   */
  get estimatedCost() {
    return this._data.estimated_cost;
  }

  /**
   * Set estimated cost
   */
  set estimatedCost(value) {
    this._data.estimated_cost = value;
  }

  /**
   * Get order ID
   */
  get orderId() {
    return this._data.order_id;
  }

  /**
   * Get form ID
   */
  get formId() {
    return this._data.form_id;
  }

  /**
   * Create a diagnosis object from API response
   * @param {Object} data - API data
   */
  static fromApiResponse(data) {
    return new Diagnosis({
      diagnosis_id: data.diagnosis_id,
      problem: data.problem,
      estimated_cost: data.estimated_cost,
      order_id: data.order_id,
      form_id: data.form_id
    });
  }
}
